dali/pipeline/data/tensor_vector.h

```diff
@@ -123,368 +92,110 @@ class TensorVector {
   }
 
   auto size() const noexcept {
-    return tensors_.size();
+    return curr_tensors_size_;
   }
 
   size_t ntensor() const noexcept {
-    return tensors_.size();
+    return curr_tensors_size_;
   }
 
-  size_t nbytes() const noexcept {
-    if (state_ == State::contiguous) {
-      return tl_->nbytes();
-    }
-    // else
-    size_t total_nbytes = 0;
-    for (const auto &t : tensors_) {
-      total_nbytes += t->nbytes();
-    }
-    return total_nbytes;
-  }
+  size_t nbytes() const noexcept;
+
+  size_t capacity() const noexcept;
+
+  TensorListShape<> shape() const;
 
-  size_t capacity() const noexcept {
-    if (state_ == State::contiguous) {
-      return tl_->capacity();
-    }
-    // else
-    size_t total_capacity = 0;
-    for (const auto &t : tensors_) {
-      total_capacity += t->capacity();
-    }
-    return total_capacity;
+  TensorShape<> tensor_shape(int idx) const {
```

Comment:
```suggestion
  const TensorShape<> &tensor_shape(int idx) const {
```

Reply:
TensorList returns by value, so I guess we should keep them consistent. If we would like to go to `const&`, that'd be out of scope of this PR.

Reply:
Because it doesn't have them. Please change it.

Reply:
Tensors can return const-ref and we shouldn't force an extra copy if we don't have to. We're not implementing a virtual interface here, more like a concept.